### PR TITLE
Allow disabling fsync to make tests run faster

### PR DESCRIPTION
**Pull request by rtm0**
### Describe Your Changes

fsync() ensures that the data is written to disk. In production this is needed for data durability. However, during the development, when the unit tests are run, this level of durability is not needed. Therefore fsync() can be disabled which will makes test runs two times faster.

The disabling is done by setting the `DISABLE_FSYNC_FOR_TESTING` environment variable. The valid values for this variable are the same as the values of the arg of `go doc strconv.ParseBool`:

```
1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False.
```

Any other value means `false`.

The variable is set for all test build targets. Compare running times:

Build Target | DISABLE_FSYNC_FOR_TESTING=0 | DISABLE_FSYNC_FOR_TESTING=1
----------------- | ------------------------------------------------ | -------------------------------------------------
make test | 1m5s  | 0m22s
make test-race | 3m1s | 1m42s
make test-pure | 1m7s | 0m20s
make test-full | 1m21s | 0m32s
make test-full-386 | 1m42s | 0m36s

When running tests for a given package, fsync can be disabled as follows:

```shell
DISABLE_FSYNC_FOR_TESTING=1 go test ./lib/storage
```

Disabling fsync() is intended for testing purposes only and the name of the variables reflects that.

What could also have been done but haven't:

-   lib/filestream/filestream.go: `Writer.MustFlush()` also uses f.Sync() but nothing has been done to it, because the Writer.MustFlush() is not used anywhere in the VM codebase. A side question: what is the general policy for the unused code?
-   lib/filestream/filestream.go: Writer.Write() calls `adviceDontNeed()` which calls unix.Fdatasync(). Disabling it could potentially improve running time, but running tests with this code disabled has shown otherwise.

### Checklist

The following checks are **mandatory**:

- [ x] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,3 +26,4 @@ _site
 /docs/.jekyll-metadata
 coverage.txt
 cspell.json
+*~
```
Comment (hagen1778) [inline .gitignore:29]: Why this change is needed?

**Review by hagen1778 — APPROVED** [review]
I'm in favour of this change. Durability isn't needed during tests and it indeed speeds up tests execution at least twice. If no objections from @f41gh7 @valyala I'd like to merge it.

File: .gitignore (hunk 1 above)
Comment (rtm0) [reply .gitignore:29]: This is to exclude emacs backup files. I use this text editor for development.
